File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlserver/DateTimeTypes.java
package dev.typr.foundations.docs.sqlserver;

import dev.typr.foundations.SqlServerType;
import dev.typr.foundations.SqlServerTypes;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.OffsetDateTime;

@SuppressWarnings("unused")
public class DateTimeTypes {
  // start
  SqlServerType<LocalDate> dateType = SqlServerTypes.date;
  SqlServerType<LocalTime> timeType = SqlServerTypes.time;
  SqlServerType<LocalTime> time3 = SqlServerTypes.timeOf(3); // TIME(3)

  // Legacy types
  SqlServerType<LocalDateTime> datetimeType = SqlServerTypes.datetime;
  SqlServerType<LocalDateTime> smalldtType = SqlServerTypes.smalldatetime;

  // Modern types (recommended)
  SqlServerType<LocalDateTime> datetime2Type = SqlServerTypes.datetime2;
  SqlServerType<LocalDateTime> datetime2_3 = SqlServerTypes.datetime2Of(3);

  // Timezone-aware
  SqlServerType<OffsetDateTime> dtoType = SqlServerTypes.datetimeoffset;
  SqlServerType<OffsetDateTime> dto3 = SqlServerTypes.datetimeoffsetOf(3);
  // stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/sqlserver/DateTimeTypes.kt
package dev.typr.foundationskt.docs.sqlserver

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import java.time.OffsetDateTime

@Suppress("unused")
class DateTimeTypes {
    //start
    val dateType: SqlServerType<LocalDate> = SqlServerTypes.date
    val timeType: SqlServerType<LocalTime> = SqlServerTypes.time
    val time3: SqlServerType<LocalTime> = SqlServerTypes.timeOf(3)  // TIME(3)

    // Legacy types
    val datetimeType: SqlServerType<LocalDateTime> = SqlServerTypes.datetime
    val smalldtType: SqlServerType<LocalDateTime> = SqlServerTypes.smalldatetime

    // Modern types (recommended)
    val datetime2Type: SqlServerType<LocalDateTime> = SqlServerTypes.datetime2
    val datetime2_3: SqlServerType<LocalDateTime> = SqlServerTypes.datetime2Of(3)

    // Timezone-aware
    val dtoType: SqlServerType<OffsetDateTime> = SqlServerTypes.datetimeoffset
    val dto3: SqlServerType<OffsetDateTime> = SqlServerTypes.datetimeoffsetOf(3)
    //stop
}
